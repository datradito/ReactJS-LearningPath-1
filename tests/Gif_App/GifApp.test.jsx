/**
 * @jest-environment jsdom
 */

import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import GifApp  from '../../src/tests/Gif_App/GifApp';

describe('Pruebas de GifApp', () => {
    test('Debe hacer match con el snapshot', () => {
        const { container } = render( <GifApp /> );
        expect( container ).toMatchSnapshot( );
    });
    test('Debe hacer la búsqueda del input', () => {
        render( <GifApp /> );
        const input = screen.getByRole('textbox');
        fireEvent.input(input, { target: {value: 'Naruto'} });
    });
    test('handleCategoryAdd debe capitalizar las búsquedas', () => {
        render( <GifApp /> );
        const input = screen.getByRole('textbox');
        fireEvent.input(input, { target: {value: 'oNe pIeCe'} });
        
        const form = screen.getByRole('form');
        fireEvent.submit(form);
        
        expect(screen.getByText('One Piece').toBeTruthy)
    });
}) 
