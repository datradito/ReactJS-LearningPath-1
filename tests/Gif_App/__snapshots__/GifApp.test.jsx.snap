// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de GifApp Debe hacer match con el snapshot 1`] = `
<div>
  <section>
    <h2>
      Trabajo de la secci√≥n 6
    </h2>
    <h3>
      Gif browser
    </h3>
  </section>
  <section>
    <form
      aria-label="form"
    >
      <input
        id="searchBar"
        name="searchBar"
        placeholder="Buscar gifs"
        type="text"
        value=""
      />
    </form>
    <div>
      <h3 />
      <h3>
        Cargando...
      </h3>
      <div
        class="card-grid"
      />
    </div>
  </section>
</div>
`;
